docs/docs/spaces.mdx

```diff
@@ -0,0 +1,237 @@
+    nlu_test: pay_cc/nlu/test_data.yml
+    rules: pay_cc/rules.yml
+    entry_intents:
+      - pay_cc
+
+importers:
+  - name: "rasa_plus.spaces.space_data_importer.SpaceDataImporter"
+    temporary_working_directory: spaces
+
+pipeline:
+  - name: WhitespaceTokenizer
+  - name: RegexFeaturizer
+  - name: LexicalSyntacticFeaturizer
+  - name: CountVectorsFeaturizer
+  - name: CountVectorsFeaturizer
+    analyzer: char_wb
+    min_ngram: 1
+    max_ngram: 4
+  - name: rasa_plus.spaces.components.spaces_crf_entity_extractor.SpacesCRFEntityExtractor
+  - name: EntitySynonymMapper
+  - name: DIETClassifier
+    epochs: 100
+    ranking_length: 0
+    entity_recognition: false
+    BILOU_flag: false
+  - name: rasa_plus.spaces.components.filter_and_rerank.FilterAndRerank
+  - name: FallbackClassifier
+    threshold: 0.3
+    ambiguity_threshold: 0.1
+
+policies:
+  - name: RulePolicy
+
+```
+
+In the above config file, you can see the spaces definition with the main space, which
+is a designated name for a space for shared primitives, and the three specialized
+spaces for transferring money, investment, and credit card payments. Each path value
+can either point to a single file or a directory.
+
+Second, you can also see the importer
+`rasa_plus.spaces.space_data_importer.SpaceDataImporter` being used. This importer
+reads the above spaces definition and assembles the bot. The results can be seen in the
+temporary working directory.
+
+Third, you can see the specific NLU components that are necessary to make spaces work.
+The most important is the `FilterAndRerank` component which is central to make spaces
+work by post-processing intent rankings and entity extractions. If you also need entity
+extraction you need to use the `SpacesCRFEntityExtractor` to make that work with spaces.
+
+Finally, you can see that we are only using the `RulePolicy` here.
+
+After you have created your spaces and made adjustments to the `config.yml`, you can
+train your new assistant using `rasa train -c config.yml` and start it afterward with
+`rasa shell`.
+
+
+## How do spaces work?
+In the following we'll look at how things work under the hood in spaces to give you a
+better understanding of what is happening in case you should need it.
+
+One of the most aspects is that spaces form a hierarchy with two layers. At the top
+is the main space which contains shared primitives that should be usable by all spaces.
+The main space is always active. Anything that is defined in a subspace, can
+only be used by that space and not by other subspaces or the main space.
+
+<img alt="two layer space hierarchy" src={useBaseUrl("/img/spaces_hierarchy.png")} />
+
+### What happens during the assembly?
+
+The most important step during the assembly is the prefixing. During this step every
+intent, entity, slot, action, form, utterance that is defined in a space's domain file
+is prefixed (infixed for utterances) with this space's name. For example, an intent
```

Comment:
I'm not sure what "(infixed for utterances)" means

Reply:
Agree, prefixing of utterances should be shown in detail (to avoid confusion during log reading, for example)